doc/distro-integration.md

```diff
@@ -0,0 +1,42 @@
+# Distro Integration
+
+Ignition is designed to be run from an initramfs, only on first boot. Parts of Ignition need to run before the rootfs is mounted and parts need to run after. Ignition requires systemd be running in the initramfs as well. This makes integration with distros somewhat complicated. As an example, the configuration for integrating Ignition with [Fedora CoreOS](fcos) can be found in the [ignition-dracut](ignition-dracut) repo.
```

Comment:
Why is `ignition-dracut` not part of this repo? They're pretty tightly coupled as it is...

Reply:
I'm hoping we can do this, but there are still some things to split out first. See https://github.com/coreos/fedora-coreos-config/pull/70#issuecomment-476642689 and https://github.com/coreos/ignition-dracut/pull/76.

Reply:
One thing I want to make clear is that using dracut is _a way_ of including it in initramfs's, not _the way_ of including it. We need to make sure the ignition-dracut modules do not have any distro-specific behavior as well.